src/lib.rs

```diff
@@ -300,6 +323,51 @@ impl RetryPolicy {
         self.retry_if(task, Always).await
     }
 
+    /// Retries a successful `Future` with this policy's configuration under certain provided conditions
+    pub async fn retry_with<T, C, S>(
+        &self,
+        task: T,
+        condition: C,
+        start_value: S,
+    ) -> Result<Vec<T::Item>, T::Error>
+    where
+        T: TaskWithParameter<S>,
+        C: SuccessCondition<T::Item, S>,
+    {
+        let mut backoffs = self.backoffs();
+        let mut condition = condition;
+        let mut task = task;
+        let mut results = vec![];
+        let mut input = start_value;
+
+        loop {
+            match task.call(input).await {
+                Ok(result) => {
+                    let new_input = condition.retry_with(&result);
+                    results.push(result);
+
+                    if new_input.is_some() {
+                        if let Some(delay) = backoffs.next() {
+                            #[cfg(feature = "log")]
+                            {
+                                log::trace!(
+                                    "task succeeded and condition is met. will try again in {:?}",
+                                    delay
+                                );
+                            }
+                            let _ = Delay::new(delay).await;
+                            input = new_input.unwrap();
```

Comment:
Could we rearrange this to avoid the unwrap? On suggestions would be an `if let Some(input) = new_input { ... }`

Reply:
Fixed.